### PR TITLE
btl_state_update_partner_menu

### PR DESCRIPTION
**Pull request by Unnunu**

### CODE REVIEW

**Comment by BowserSlug** [comment]
🚀 This PR matches 1 function (+0.17%) on `us`.

File: include/variables.h [hunk 1]
```diff
@@ -21,7 +21,7 @@ extern BattleStatus gBattleStatus;
 extern s32 D_800DC4D0;
 extern s32 D_800DC4D4;
 extern s32 D_800DC4D8;
-extern s32 gBattleState2;
+extern s32 gBattleSubState;
```
Comment (nanaian) [inline include/variables.h:24]: Can you rename the enum and its prefix also?

File: src/415D90.c [hunk 2]
```diff
@@ -14,6 +14,73 @@
 #include "sprite/npc/BattleLakilester.h"
 #include "sprite/npc/Twink.h"
 
+enum BattleSubStatesPartner {
```
Comment (nanaian) [inline src/415D90.c:17]: Move to enums.h or 415D90.h please

File: src/415D90.c [hunk 3]
```diff
@@ -3534,60 +3610,864 @@ void btl_state_draw_player_menu(void) {
...
+        battle_menu_moveOptionCount = battleStatus->submenuMoveCount;
+        s1 = battleStatus->unk_5C[6];
+        D_802AD4A8 = s1;
+        func_802A2684();
+        gBattleSubState = BATTLE_SUB_STATE_PARTNER_FOCUS_2;
+        break;
+    case BATTLE_SUB_STATE_PARTNER_FOCUS_2:
+        numOptions = func_802A2C84();
+        if (numOptions != 0) {
+            if (numOptions == 255) {
```
Comment (nanaian) [inline src/415D90.c:4167]: Interesting this isn't -1. Does changing `numOptions` to 8-bit allow -1 here and not break elsewhere?

**Review by nanaian — CHANGES_REQUESTED** [review]
Awesome, mad that this func was 0.17%. Couple comments

File: src/415D90.c (hunk 3 above)
Comment (Unnunu) [reply src/415D90.c:4100]: -1 and 255 are different possible results that can be returned by this kind of function. 255 means return back from submenu and -1 is for error.

**Pull request merged**